docs/DeveloperGuide.adoc

```diff
@@ -520,35 +520,33 @@ The List mechanism is facilitated by commands which use `Logic` interface to obt
+    model.updateFilteredBlacklistedPersonList(PREDICATE_SHOW_ALL_BLACKLISTED_PERSONS);
+    String currentList = ListObserver.getCurrentListName();
+    return new CommandResult(currentList + MESSAGE_SUCCESS);
 }
 ----
 
 Specifically for `blacklist` command, it calls the `getFiltererdBlacklistedPersons()` method residing in the `Logic` interface. `LogicManager` subsequently calls `Model` interface. `Model` uses `ModelManager` class to handle the command and thereafter calls `ReadOnlyAddressBook` interface to handle the request. `ReadOnlyAddressBook` uses `AddressBook` class to handle this request.
 
-In the `AddressBook` class, there is only one `persons` variable that stores all `ReadOnlyPerson` class objects. The blacklisted persons are obtained by running a check on all debtors residing in this variable. The check is executed using the `ReadOnlyPerson#getIsBlacklisted()` method, as shown below:
+In the `AddressBook` class, there is only one `persons` variable that stores all `ReadOnlyPerson` class objects. The blacklisted persons are obtained by running a check on all debtors residing in this variable. The check is executed using the `asObservableBlacklist()` method, as shown below:
+
 [source,java]
 ----
-public UniquePersonList getBlacklistedPersons() {
-    UniquePersonList blacklistedPersons = new UniquePersonList();
-    for (Person person : persons.getInternalList()) {
-        if (person.getIsBlacklisted()) {
-            try {
-                blacklistedPersons.add(person);
-            } catch (DuplicatePersonException e) {
-                assert false : "This is not possible as prior checks have been done";
-            }
-        }
-    }
-    return blacklistedPersons;
+public ObservableList<ReadOnlyPerson> asObservableBlacklist() {
+    return FXCollections.unmodifiableObservableList(mappedList.stream()
+            .filter(person -> person.isBlacklisted()).collect(toCollection(FXCollections::observableArrayList)));
 }
 ----
 
 Although this way of implementation seems inefficient, it supports robust synchronisation among the various other lists. For example, if a person is deleted from the masterlist, he will also be deleted from the blacklist. Likewise for various other commands that changes the `ReadOnlyPerson` object. Thus, it is efficient in this aspect. Moreover, this implementation sets the groundwork for future implementations of various other lists.
 
+One other important aspect to this implementation is that commands now have exclusive control over the list. If the user wishes to only work with blacklist, it is very much possible and all the current commands will only work on blacklisted persons. The persons not in this list will not be affected. For example, find command will only search among the blacklisted persons. This applies to all other commands. Nonetheless, if a person object is modified, the modification will be reflected across all other lists, given that the person exists in these various lists.
```

Comment:
I don't see how the explanation is for "exclusive control over the list", what does "exclusive control over the list" even mean?
`find` command
I understand what "this" refers to in "This applies to all other commands" but it may not be intuitive for everyone reading this

Reply:
Noted